hack/update-golangci-lint-config.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script checks that golangci-strict.yaml and golangci.yaml remain in
+# sync. Lines that are intentionally different must have a comment which
+# mentions golangci.yaml or golangci-lint.yaml.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+# This sets up the environment, like GOCACHE, which keeps the worktree cleaner.
+kube::golang::setup_env
+
+# Remove all files, some of them might be obsolete.
+rm -f hack/golangci*.yaml
+
+generate () {
+    out="$1"
+    shift
+    echo "Generating $out from hack/golangci.yaml.in with ./cmd/gotemplate $*"
+    go run ./cmd/gotemplate <hack/golangci.yaml.in >"${out}" "$@"
+}
+
+# Regenerate.
+generate hack/golangci.yaml Base=1
+generate hack/golangci-strict.yaml Strict=1
+generate hack/golangci-hints.yaml Hints=1
+
+# Inform about differences.
+echo
+diff -c hack/golangci.yaml hack/golangci-strict.yaml || true
```

Comment:
If we are going to diff, can we use `-u` ?

Reply:
Changed. Here's what an invocation now looks like:

```console
$ hack/update-golangci-lint-config.sh 
+++ [0621 08:45:09] Setting GOMAXPROCS: 36
Generating hack/golangci.yaml from hack/golangci.yaml.in with ./cmd/gotemplate Base=1
Generating hack/golangci-strict.yaml from hack/golangci.yaml.in with ./cmd/gotemplate Strict=1
Generating hack/golangci-hints.yaml from hack/golangci.yaml.in with ./cmd/gotemplate Hints=1

--- hack/golangci.yaml	2023-06-21 08:45:10.222186135 +0200
+++ hack/golangci-strict.yaml	2023-06-21 08:45:10.478185292 +0200
@@ -30,11 +30,6 @@
     - path: conversion\.go
       linters:
         - ineffassign
-    # TODO(oscr) Remove these excluded directories and fix findings. Due to large amount of findings in different components
-    # with different owners it's hard to fix everything in a single pr. This will therefore be done in multiple prs.
-    - path: (pkg/volume/*|test/*|azure/*|pkg/cmd/wait*|request/bearertoken/*|metrics/*|filters/*)
-      linters:
-        - gocritic
 
     # The Kubernetes naming convention for conversion functions uses underscores
     # and intentionally deviates from normal Go conventions to make those function
@@ -88,7 +83,7 @@
       text: "ST1023: should omit type .* from declaration; it will be inferred from the right-hand side"
 
 linters:
-  disable-all: true
+  disable-all: false
   enable: # please keep this alphabetized
     - ginkgolinter
     - gocritic
@@ -98,6 +93,9 @@
     - staticcheck
     - stylecheck
     - unused
+  disable:
+    # https://github.com/kubernetes/kubernetes/issues/117288#issuecomment-1507008359
+    - errcheck
 
 linters-settings: # please keep this alphabetized
   custom:
@@ -106,15 +104,6 @@
       path: ../_output/local/bin/logcheck.so
       description: structured logging checker
       original-url: k8s.io/logtools/logcheck
-  gocritic:
-    enabled-checks:
-      - equalFold
-      - boolExprSimplify
   staticcheck:
     checks:
       - "all"
-      - "-SA1019"  # TODO(fix) Using a deprecated function, variable, constant or field
-      - "-SA2002"  # TODO(fix) Called testing.T.FailNow or SkipNow in a goroutine, which isn’t allowed
-  stylecheck:
-    checks:
-      - "ST1019"   # Importing the same package multiple times

--- hack/golangci-strict.yaml	2023-06-21 08:45:10.478185292 +0200
+++ hack/golangci-hints.yaml	2023-06-21 08:45:10.734184449 +0200
@@ -46,42 +46,6 @@
         - gocritic
       text: "ifElseChain: rewrite if-else to switch statement"
 
-    # The following issues were deemed "might be worth fixing, needs to be
-    # decided on a case-by-case basis".  This was initially decided by a
-    # majority of the developers who voted in
-    # https://github.com/kubernetes/kubernetes/issues/117288 and may evolve
-    # over time.
-
-    # https://github.com/kubernetes/kubernetes/issues/117288#issuecomment-1507008918
-    - linters:
-        - gocritic
-      text: "assignOp:"
-
-    # https://github.com/kubernetes/kubernetes/issues/117288#issuecomment-1507016854
-    - linters:
-        - gosimple
-      text: "S1002: should omit comparison to bool constant"
-
-    # https://github.com/kubernetes/kubernetes/issues/117288#issuecomment-1507023980
-    - linters:
-        - gosimple
-      text: "S1016: should convert opts .* instead of using struct literal"
-
-    # https://github.com/kubernetes/kubernetes/issues/117288#issuecomment-1507026758
-    - linters:
-        - gosimple
-      text: "S1033: unnecessary guard around call to delete"
-
-    # https://github.com/kubernetes/kubernetes/issues/117288#issuecomment-1507030071
-    - linters:
-        - stylecheck
-      text: "ST1012: error var .* should have name of the form ErrFoo"
-
-    # https://github.com/kubernetes/kubernetes/issues/117288#issuecomment-1507031224
-    - linters:
-        - stylecheck
-      text: "ST1023: should omit type .* from declaration; it will be inferred from the right-hand side"
-
 linters:
   disable-all: false
   enable: # please keep this alphabetized
@@ -93,9 +57,6 @@
     - staticcheck
     - stylecheck
     - unused
-  disable:
-    # https://github.com/kubernetes/kubernetes/issues/117288#issuecomment-1507008359
-    - errcheck
 
 linters-settings: # please keep this alphabetized
   custom:
```

Is that perhaps too verbose?

Reply:
Diff is removed now.